src/aspire/basis/fb_2d.py

```diff
@@ -220,32 +218,28 @@ def _evaluate(self, v):
 
         return x
 
-    def _evaluate_t(self, v):
+    def _evaluate_t(self, x):
         """
         Evaluate coefficient in FB basis from those in standard 2D coordinate basis
 
-        :param v: The coefficient array to be evaluated. The last dimensions
+        :param x: The coefficient array to be evaluated. The last dimensions
             must equal `self.sz`.
         :return: The evaluation of the coefficient array `v` in the dual basis
             of `basis`. This is an array of vectors whose last dimension equals
             `self.count` and whose first dimensions correspond to
             first dimensions of `v`.
         """
-        v = v.T
-        x, sz_roll = unroll_dim(v, self.ndim + 1)
-        x = m_reshape(
-            x, new_shape=tuple([np.prod(self.sz)] + list(x.shape[self.ndim :]))
-        )
+        x = x.reshape(x.shape[0], -1)
```

Comment:
Looks like the stack shape is lost here (it's not saved or restored later in `v`).

Reply:
Yeah, I had the same thought, but the stack is handled by the public method `evaluate_t`. It's actually passed to this private method already stack-flattened.

Reply:
Ok got it.